aten/src/ATen/native/xnnpack/Factory.cpp

```diff
@@ -21,15 +26,28 @@ Tensor empty_with_tail_padding(
           c10::Storage{
               dtype,
               nelements,
-              allocator.allocate(nelements * dtype.itemsize()),
-              &allocator,
+              allocator_ptr->allocate(nelements * dtype.itemsize()),
+              allocator_ptr,
               /*resizable=*/true,
           },
           DispatchKeySet{DispatchKey::CPUTensorId}));
 
   return tensor.resize_(size, memory_format);
 }
 
+Tensor allocate_padded_if_needed(const Tensor& input_contig) {
```

Comment:
@dzhulgakov, I think the current approach will work with views as well, since the guarding allocator guarantees some number of safe-to-access bytes beyond the end of the tensor (which is the requirement for xnnpack). However, I am not sure how to use `storage_offset + nbytes` and `storage().capacity()` to deduce that there are some  safe-to-access bytes allocated beyond the end of the tensor. I think only the allocator knows truly how many bytes it allocated. Please correct me if I am wrong.
Honestly, it would be great if we can determine 1) whether tensor dimensions are such that no out-of-bound access will occur. or 2) if guarding allocator is used. 2 we covered. But if we cover 1 as well, it will greatly simplify allocations and for the most use cases I think 1 should be satisfied.

Reply:
I think we should stick with this version now because it is simple and safe.  We can follow up to discuss Dima's suggestion later.